docs/datamaps/index.md

```diff
@@ -0,0 +1,169 @@
+
+You can think of tags as entry->boolean maps, while data maps are more flexible entry->object maps.
+
+A data map can be attached to both static, built-in, registries and dynamic data-driven datapack registries.  
+
+Data maps support reloading through the use of the `/reload` command or any other means that reload server resources.
+
+## Registration
+A data map type should be statically created and then registered to the `RegisterDataMapTypesEvent` (which is fired on the mod event bus). The `DataMapType` can be created using a `DataMapType$Builder`, through `DataMapType#builder`.  
+
+The builder provides a `syced` method which can be used to mark a data map as synced and have it sent to clients.  
+
+A simple `DataMapType` has two generic arguments: `R` (the type of the registry the data map is for) and `T` (the values that are being attached). A data map of `SomeObject`s that are attached to `Item`s can, as such, be represented as `DataMapType<Item, SomeObject>`.  
+
+Data maps are serialized and deserialized using [Codecs](../datastorage/codecs.md).
+
+Let's take the following record representing the data map value as an example:
+```java
+public record DropHealing(
+        float amount, float chance
+) {
+    public static final Codec<DropHealing> CODEC = RecordCodecBuilder.create(in -> in.group(
+            Codec.FLOAT.fieldOf("amount").forGetter(DropHealing::amount),
+            Codec.floatRange(0, 1).fieldOf("chance").forGetter(DropHealing::chance)
+    ).apply(in, DropHealing::new));
+}
+```
+
+:::warning
+The value (`T`) should be an *immutable* object, as otherwise weird behaviour can be caused if the object is attached to all entries within a tag (since no copy is created).
+:::
+
+For the purposes of this example, we will use this data map to heal players when they drop an item.  
+The `DataMapType` can be created as such:
+```java
+public static final DataMapType<Item, DropHealing> DROP_HEALING = DataMapType.builder(
+        new ResourceLocation("mymod:drop_healing"), Registries.ITEM, DropHealing.CODEC
+).build();
+```
+and then registered to the `RegisterDataMapTypesEvent` using `RegisterDataMapTypesEvent#register`.
+
+## Syncing
+A synced data map will have its values synced to clients. A data map can be marked as synced using `DataMapType$Builder#synced(Codec<T> networkCodec, boolean mandatory)`.   
+The values of the data map will then be synced using the `networkCodec`.  
+If the `mandatory` flag is set to `true`, clients that do not support the data map (including Vanilla clients) will not be able to connect to the server, nor vice-versa. A non-mandatory data map is, on the other side, optional, so it will not prevent any clients from joining.
+
+## JSON Structure and location
+Data maps are loaded from a JSON file located at `:mapNamespace/data_maps/:registryNamespace/:registryPath/:mapPath.json`.  
```

Comment:
I understand the parameters, but if you're going to talk about them here, they should also be defined here.